python/pyspark/sql/readwriter.py

```diff
@@ -282,6 +282,23 @@ def parquet(self, *paths):
         """
         return self._df(self._jreader.parquet(_to_seq(self._spark._sc, paths)))
 
+    @since(2.2)
+    def parquet(self, path):
```

Comment:
Having two functions with the same name and different args doesn't behave like in Scala (so this won't work). Please use kwargs or similar and add a test for paths and path.